ete4/coretype/tree.pyx

```
@@ -882,7 +882,7 @@ cdef class Tree(object):
         :param topological: If True, distance will refer to the number of
             nodes between target and target2.
         """
-        d = (lambda node: 1) if topological else (lambda node: node.dist)
+        d = (lambda node: 1) if topological else (lambda node: node.dist or 0.0)
```

Uhm, this also seems unrelated to set_outgroup() (so, not for this PR?).

But it also seems incorrect to return a distance if one of the branches does not have it defined. After all, None is not the same as 0, and I think `get_distance()` should fail in that case rather than ignore that branch, no?